debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ApicurioInfrastructure.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.testing.testcontainers;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.lifecycle.Startables;
+
+import java.util.stream.Stream;
+
+/**
+ * Apicurio Testcontainers infrastructure handling.
+ */
+
+public class ApicurioInfrastructure extends GenericContainer<ApicurioInfrastructure> {
+
+    protected static final String DEBEZIUM_VERSION = "1.3.0.Final";
+    protected static final String APICURIO_VERSION = "1.3.2.Final";
+    private final GenericContainer<?> apicurioContainer;
+    private final ImageFromDockerfile apicurioContainerImage;
+
+    private ApicurioInfrastructure() {
+        apicurioContainerImage = new ImageFromDockerfile()
+                .withDockerfileFromBuilder(builder -> builder
+                        .from("debezium/connect:" + DEBEZIUM_VERSION)
+                        .env("KAFKA_CONNECT_DEBEZIUM_DIR", "$KAFKA_CONNECT_PLUGINS_DIR/debezium-connector-postgres")
+                        .env("APICURIO_VERSION", APICURIO_VERSION)
+                        .run("cd $KAFKA_CONNECT_DEBEZIUM_DIR && curl https://repo1.maven.org/maven2/io/apicurio/apicurio-registry-distro-connect-converter/$APICURIO_VERSION/apicurio-registry-distro-connect-converter-$APICURIO_VERSION-converter.tar.gz | tar xzv")
+                        .build());
+
+        apicurioContainer = new GenericContainer<>("apicurio/apicurio-registry-mem:" + APICURIO_VERSION)
+                .withExposedPorts(8080)
+                .waitingFor(new LogMessageWaitStrategy().withRegEx(".*apicurio-registry-app.*started in.*"));
+
+    }
+
+    public static ApicurioInfrastructure getDebeziumApicurioDefaultInfrastructure() {
+        return new ApicurioInfrastructure();
+    }
+
+    public GenericContainer<?> getApicurioContainer() {
+        return apicurioContainer;
+    }
+
+    public ImageFromDockerfile getApicurioContainerImage() {
+        return apicurioContainerImage;
+    }
+
+    public void startContainer() {
+        Startables.deepStart(Stream.of(apicurioContainer)).join();
```

Comment:
This can be removed. Responsibility of Infrastructure classes is only providing the infrastructure (pre-configured containers) the lifecycle is handled outside (in test classes etc).